control_plane/attachment_service/src/service.rs

```diff
@@ -444,11 +478,14 @@ impl Service {
         let result_task_this = this.clone();
         tokio::task::spawn(async move {
             while let Some(result) = result_rx.recv().await {
-                tracing::info!(
-                    "Reconcile result for sequence {}, ok={}",
-                    result.sequence,
-                    result.result.is_ok()
+                let span = tracing::span!(tracing::Level::INFO, "reconcile_result",
+                        tenant_id=%result.tenant_shard_id.tenant_id,
+                        shard_id=%result.tenant_shard_id.shard_slug(),
+                        sequence=%result.sequence,
                 );
+                let _span = span.enter();
```

Comment:
Please always use `.instrument`. It's correct right now because we have no .await points in this block, but, someone could add them in the future and they will forget to change to instrument then.
Yes, it will make the code more ugly and add another indent level, but, that's the price we pay.

https://docs.rs/tracing/latest/tracing/struct.Span.html#in-asynchronous-code

Reply:
This was a bit of a drive-by thing on this PR, but it's a good time to refactor the per-tenant part into a function anyway so that it can have an #instrument: e68a8d10e